test/results_test.dart

```diff
@@ -138,6 +138,21 @@ Future<void> main([List<String>? args]) async {
     realm.close();
   });
 
+  test('Results requested wrong index throws', () {
+    var config = Configuration([Car.schema]);
+    var realm = Realm(config);
+
+    final car = Car('');
+    realm.write(() => realm.add(car));
+
+    final cars = realm.all<Car>();
+    realm.write(() => realm.deleteMany(cars));
+
+    expect(() => cars[0], throws<RealmException>("Requested index 0 in empty Results"));
```

Comment:
Why not just test this after opening the Realm since its empty we could skip adding objects to the realm

Reply:
Done